src/main/java/com/dpgil/pathlinker/path_linker/internal/PathLinkerResultPanel.java

```diff
@@ -0,0 +1,377 @@
+		constraint.anchor = GridBagConstraints.LINE_START;
+		constraint.weightx = 1;
+		constraint.gridx = 0;
+		constraint.gridy = 1;
+		constraint.gridwidth = 2;
+
+		this.add(scrollPane, constraint);
+	}
+
+	/**
+	 * The method is triggered by DownloadButtonListener
+	 * Creates a dialogue for user to save the result tables
+	 * @throws IOException
+	 */
+	private void downloadResultTable() throws IOException {
+		// override approveSelection method to warn if user overwrites a existing file when saving
+		JFileChooser fc = new JFileChooser() {
+			@Override
+			public void approveSelection(){
+				File f = getSelectedFile();
+				if (f.exists() && getDialogType() == SAVE_DIALOG) {
+					int result = JOptionPane.showConfirmDialog(this,
+							"The file exists, overwrite?","Existing file",JOptionPane.YES_NO_CANCEL_OPTION);
+					switch(result) {
+					case JOptionPane.YES_OPTION:
+						super.approveSelection();
+						return;
+					case JOptionPane.NO_OPTION:
+						return;
+					case JOptionPane.CLOSED_OPTION:
+						return;
+					case JOptionPane.CANCEL_OPTION:
+						cancelSelection();
+						return;
+					}
+				}
+				super.approveSelection();
+			}
+		};
+
+		fc.setDialogTitle("Sepcify the file to save"); // title of the dialogue
+		int userSelection = fc.showSaveDialog(null);
+
+		if (userSelection == JFileChooser.APPROVE_OPTION) {
+			File fileToSave = fc.getSelectedFile();
+			BufferedWriter writer = new BufferedWriter(new FileWriter(fileToSave.getAbsolutePath()));
+
+			for(int i = 0; i < _resultTable.getColumnCount(); i++) {
+				writer.write(_resultTable.getColumnName(i));
+				writer.write("\t");
+			}
+
+			for (int i = 0; i < _resultTable.getRowCount(); i++) {
+				writer.newLine();
+				for(int j = 0; j < _resultTable.getColumnCount(); j++) {
+					writer.write((_resultTable.getValueAt(i, j).toString()));
+					writer.write("\t");;
+				}
+			}
+
+			writer.close();
+			JOptionPane.showMessageDialog(null, "Download successful");
+		} else {
+			JOptionPane.showMessageDialog(null, "Download canceled");
+		}
+	}
+
+	/**
+	 * Converts a path to a string concatenating the node names A path in the
+	 * network involving A -> B -> C would return A|B|C
+	 *
+	 * @param p
+	 *            the path to convert to a string
+	 * @return the concatenation of the node names
+	 */
+	private String pathAsString(Path p)
+	{
+		// builds the path string without supersource/supertarget [1,len-1]
+		StringBuilder currPath = new StringBuilder();
+		for (int i = 1; i < p.size() - 1; i++)
+			currPath.append(p.nodeIdMap.get(p.get(i)) + "|");
+
+		currPath.setLength(currPath.length() - 1);
+
+		return currPath.toString();
+	}
+
+	@Override
+	public Component getComponent() {
+		return this;
+	}
+
+	@Override
+	public CytoPanelName getCytoPanelName() {
+		return CytoPanelName.EAST;
+	}
+
+	@Override
+	public Icon getIcon() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public String getTitle() {
+		return "Result " + _title;
```

Comment:
Also, the index of the result (such as Result 1 or Result 4) should match the index of the "path index" column (such as path index 1 or path index 4) in the Edge Table.